modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -3775,57 +3775,42 @@ private synchronized void scheduleNext() {
         }
 
         /** Stopping handler. */
-        public synchronized void stop() {
-            stopping = true;
-
-            if (active != null)
-                active.acceptException(new IgniteException(SNP_NODE_STOPPING_ERR_MSG));
+        public void stop() {
+            synchronized (this) {
+                stopping = true;
```

Comment:
Suggestion: comment why `sync.` instead of a `volotile` on juts one `boolean` flag: like 'prevents running `submit()`'. Up to you.

Reply:
I think this is an established and transparent way to stop processes.